File: security-adapter-in/src/main/java/com/example/security/adapter/in/rest/dto/IngestRequest.java
package com.example.security.adapter.in.rest.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.time.Instant;
import java.util.Map;

/** raw event 수집 요청. */
public record IngestRequest(
    @NotBlank String tenantId,
    @NotBlank String source,
    @NotBlank String schema,
    @NotNull Map<String, Object> payload,
    /** 클라이언트가 보고한 timestamp (없으면 서버 수신 시각 사용). */
    Instant occurredAt) {}

File: security-adapter-in/src/main/kotlin/com/example/security/adapter/in/rest/dto/IngestRequest.kt
package com.example.security.adapter.`in`.rest.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.Instant

/**
 * POST /api/v1/ingest — raw event 수집 요청.
 *
 * tenantId 는 path 가 아닌 body 에서 받는다 (multi-tenant 환경에서 운영자가 명시적으로 지정).
 */
@JvmRecord
data class IngestRequest(
    @field:NotBlank val tenantId: String,
    @field:NotBlank val source: String,
    @field:NotBlank val schema: String,
    @field:NotNull val payload: Map<String, Any?>,
    /** 클라이언트가 보고한 timestamp (없으면 서버 수신 시각 사용). */
    val occurredAt: Instant?,
)
